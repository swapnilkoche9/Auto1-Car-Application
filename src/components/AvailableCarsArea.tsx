import React from 'react'
import AvailableCar from './AvailableCar'
import Pagination from './Pagination'
import Dropdown from './Dropdown'
import constants from '../constants/SystemConstants'
import {AvailableCarsAreaProps,Car,Cars} from '../utils/interface'


const sortDropDownContent = [constants.NONE_SORT_ORDER, constants.ASC_SORT_ORDER, constants.DES_SORT_ORDER]
let renderAvailableCars = (cars:Cars) => {
  return cars && cars.map((car:Car, index:number) => {
    const {
      stockNumber,
      mileage,
      fuelType,
      color,
      modelName,
      pictureUrl
    } = car
    return (
      <AvailableCar
        key={index}
        stockNumber={stockNumber}
        mileage={mileage}
        fuelType={fuelType}
        color={color}
        modelName={modelName}
        pictureUrl={pictureUrl}
      />
    )
  })
}

const AvailableCarsArea = (props:AvailableCarsAreaProps) => {
  const {
    cars,
    totalPageCount,
    currentPage,
    changePageNumber,
    totalCarCount,
    getPageParams,
    getSortFilterParams
  } = props

  return (
    <div className='availableCarsContainer'>
      <div className="availableTextAndSortContainer">
        <div className="availableTextAndResultContainer">
          <div className="availableTextContainer">
            <p className='availableText'>{constants.AVAILABLE_CARS_TEXT}</p>
          </div>
          <div className="resultTextContainer">
            <p className='resultText'>{`${constants.SHOWING_TEXT} ${cars.length} ${constants.OF_TEXT} ${totalCarCount} ${constants.RESULT_TEXT}`}</p>
          </div>
        </div>
        <div className="sortDropDownContainer">
          <Dropdown
            dropdownContent={sortDropDownContent}
            defaultDropdownValue={constants.NONE_SORT_ORDER}
            getFilterParams={getSortFilterParams}
            dropDownLabel={constants.SORT_BY_TEXT}
          />
        </div>
      </div>
      {renderAvailableCars(cars)}
      <Pagination
        getPageParams={getPageParams}
        totalPageCount={totalPageCount}
        currentPage={currentPage}
        changePageNumber={changePageNumber}
      />
    </div>
  )
}

export default AvailableCarsArea
