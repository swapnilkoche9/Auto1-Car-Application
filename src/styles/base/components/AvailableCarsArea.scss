
.availableCarsContainer {
  width: 75%;
  padding: $xs-size;
  margin-right: $xl-size
}
.availableTextAndSortContainer{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}
.availableTextAndResultContainer{
  text-align: left;
}
.sortDropDownContainer{
  margin-top: $l-size;
  width: 300px;
}

.availableCar {
  width: 100%;
  border: 0.2rem solid $light-gray;
  margin-top: $m-size;
  height: 13rem;
  display: flex;
  flex-direction: row;
  justify-content: flex-start
}

.carImageContainer {
  margin: $m-size;
  width: 15%; 
}
.carImage{
  height: 90px;
  padding: $xs-size;
}

.carDetailsContainer {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  margin: $m-size;
  width: 85%;
}

.availableCarDetails {
  display: flex;
  flex-direction: column;
  margin: $m-size;
  width: 100%;
}

.availableText {
  font-size: $l-size;
  font-weight: 700;
  color: $dark-gray
}

.resultText {
  font-size: $l-size;
  color: $dark-gray;
}
.availableCarstext {
  text-align: start;
  margin: 0;
  color: $dark-gray;
}

.availableCarsTextBold{
  font-size: $l-size;
  font-weight: 700;
}

.availableCarsTextSmall{
  font-size: $m-size;
}
.viewDetailsLinkContainer{
  text-align: left;
}
.linkTextOange{
  color: $orange;
  text-decoration: none;
}
.linkTextOange:hover{
  color: $dark-orange;
  text-decoration: underline;
  
}


@media only screen and (max-width: 600px) {
  .availableTextAndSortContainer{
    flex-direction: column;
    margin-left: 30px;

  }
  .availableCar{
    width: max-content;
    flex-direction: column;
    height: 25rem;
    margin-left: 30px;
  }
  .carImageContainer{
    width: unset;
    margin: unset;
  }
}