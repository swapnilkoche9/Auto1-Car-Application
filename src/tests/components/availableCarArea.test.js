import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import AvailableCarsArea from '../../components/AvailableCarsArea'

Enzyme.configure({ adapter: new Adapter() })
const { shallow } = Enzyme

const getPageParams = jest.fn()
const getSortFilterParams = jest.fn()
const wrapper = shallow(
  <AvailableCarsArea
    getPageParams={getPageParams}
    getSortFilterParams={getSortFilterParams}
    cars={[{
      stockNumber: 100,
      mileage: { number: 100, unit: 'km' },
      fuelType: 'Petrol',
      color: 'Yellow',
      modelName: 'Taurus'
    }]}
    totalPageCount={100}

  />
)
test('should render AvailableCarsArea correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should render AvailableCarsArea with alt data correctly', () => {
  wrapper.setProps({
    color: 'block'
  })
  expect(wrapper).toMatchSnapshot()
})
