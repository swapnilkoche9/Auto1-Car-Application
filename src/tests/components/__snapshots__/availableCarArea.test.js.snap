// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AvailableCarsArea correctly 1`] = `ShallowWrapper {}`;

exports[`should render AvailableCarsArea with alt data correctly 1`] = `ShallowWrapper {}`;
